api/README.md

```diff
@@ -0,0 +1,20 @@
+# Elastic Agent Shipper Client API
```

Comment:
👍 This documentation is excellent, I particularly like the Technical Considerations section with all the rationale for the design.

Reply:
Julien asked for this to be part of the PR after I infodumped all the design conversations to him ^,^